src/pydata_sphinx_theme/translator.py

```diff
@@ -23,11 +24,35 @@ def __init__(self, *args, **kwds):
         self.settings.table_style = "table"
 
     def starttag(self, *args, **kwargs):
-        """Ensure an aria-level is set for any heading role."""
+        """Perform small modifications to tags.
+
+        - ensure aria-level is set for any tag with heading role
+        - ensure <pre> tags have tabindex="0".
+        """
         if kwargs.get("ROLE") == "heading" and "ARIA-LEVEL" not in kwargs:
             kwargs["ARIA-LEVEL"] = "2"
+
+        if "pre" in args:
+            kwargs["tabindex"] = "0"
+
         return super().starttag(*args, **kwargs)
 
+    def visit_literal_block(self, node):
+        """Modify literal blocks.
+
+        - ensure tabindex="0" for <pre> tags buried in the HTML tree that Sphinx
+          (with Pygments) generates for literal blocks
+        """
+        try:
+            super().visit_literal_block(node)
+        except nodes.SkipNode:
```

Comment:
possibly slightly better here is 

```suggestion
        except nodes.SkipNode as e:
```

followed by `raise e from None` on line 54.  I *think* it won't make much difference here, since the `raise` is not happening inside a second try/except. But I think it will preserve the exception's source in the traceback as having originated within docutils code, not within our code... talking it through, I'm actually not sure if that's preferable or not :/